deepchem/data/dft_data.py

```diff
@@ -0,0 +1,319 @@
+    ``Entry.create``
+    """
+
+    created_entries: Dict[str, DFTEntry] = {}
+
+    @classmethod
+    def create(cls,
+               entry_dct: Union[Dict, DFTEntry],
+               device: torch.device,
+               dtype: torch.dtype = torch.double) -> DFTEntry:
+        if isinstance(entry_dct, DFTEntry):
+            return entry_dct
+
+        s = str(entry_dct)
+        if s not in cls.created_entries:
+            tpe = entry_dct["type"]
+            kwargs = {
+                "entry_dct": entry_dct,
+                "dtype": dtype,
+                "device": device,
+            }
+            obj = {
+                "ae": EntryAE,
+                "ie": EntryIE,
+                "dm": EntryDM,
+                "dens": EntryDens,
+                "force": EntryForce,
+            }[tpe](**kwargs)
+            cls.created_entries[s] = obj
+        return cls.created_entries[s]
+
+    def __init__(self,
+                 entry_dct: Dict,
+                 device: torch.device,
+                 dtype: torch.dtype = torch.double):
+        super().__init__(entry_dct)
+        self._systems = [DFTSystem.create(p) for p in entry_dct["systems"]]
+        self._dtype = dtype
+        self._device = device
+
+        self._trueval_is_set = False
+        self._trueval = torch.tensor(0.0).to(device)
+        """
+        Parameters
+        ----------
+        systems:  List[DFTSystem]
+            Returns the list of systems in the entry
+        dtype: torch.dtype
+            Returns data type of the calculated and true energies of a system
+        device: torch.device
+        """
+
+    @property
+    def dtype(self) -> torch.dtype:
+        return self._dtype
+
+    @property
+    def device(self) -> torch.device:
+        return self._device
+
+    def get_systems(self) -> List[DFTSystem]:
+        """
+        Returns the list of systems in the entry
+        """
+        return self._systems
+
+    @abstractproperty
+    def entry_type(self) -> str:
+        """
+        Returning the type of the entry of the dataset; 
+        1) Atomic Ionization Potential (IP/IE)
+        2) Atomization Energy (AE)
+        3) Density Profile (DENS)
+        4) Density Matrix (DM)
+        """
+        pass
+
+    def get_true_val(self) -> torch.Tensor:
+        if not self._trueval_is_set:
+            self._trueval = self._get_true_val()
+            self._trueval_is_set = True
+        return self._trueval
+
+    @abstractmethod
+    def _get_true_val(self) -> torch.Tensor:
+        """
+        Get the true value of the DFTEntry.  
+        For the AE and IP entry types, the experimental values are collected from the NIST CCCBDB/ASD 
+        databases. 
+        The true values of density profiles are calculated using PYSCF-CCSD calculations. This method            simply loads the value, no calculation is performed.  
+        """
+        pass
+
+    @abstractmethod
+    def get_val(self, qcs: List[KSCalc]) -> torch.Tensor:
```

Comment:
I think you may want to have these construct numpy arrays instead of torch tensors. We don't currently have data classes that use torch tensors. With numpy arrays, we can also store them in `DiskDataset` objects to disk